src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -145,6 +183,34 @@ public bool Equals(ArraySegment<T> obj)
             return obj._array == _array && obj._offset == _offset && obj._count == _count;
         }
 
+        public ArraySegment<T> Slice(int index)
+        {
+            ThrowInvalidOperationIfDefault();
+            // Note: `index` is allowed to be negative so the start of this ArraySegment can be moved backwards.
```

Comment:
@terrajobst Do you have an opinion on this either? (This is blocking the APIs from being added.)